soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -373,4 +385,35 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 	z_xt_ints_on(core_desc[cpu].intenable);
 }
 
+#if CONFIG_ADSP_DYNAMIC_CLOCK_GATING
+
+atomic_t clock_gating_lock;
+
+void adsp_clock_gating_lock_get(void)
+{
```

Comment:
I wonder if we could have this get/put in a separate PR. 

I'm a bit worried how this going to be used. As it is here, this is very Intel specific function, so any app code using this, will be very specific to one type of hardware. OTOH, the substates were not much more generic in the end either. 

@tmleman In what case SOF would have to block clock gating? Is it just to implement the "prevent_local_clock_gating" and "prevent_power_gating" SET_D0IX IPC command fields? Or do we have some other needs? These IPCs are very Intel specific and for these the adsp_clock/power_gating_lock() interface is ok, but we need to put these into some Intel specific corner of SOF code as well.

@ceolin Do we have something in Zephyr like the QoS API in Linux, where a generic driver can set a generic constraint on responsiveness of the CPU, effectively blocking low-power states in a (still generic) manner.

Reply:
> @tmleman In what case SOF would have to block clock gating? Is it just to implement the "prevent_local_clock_gating" and "prevent_power_gating" SET_D0IX IPC command fields? Or do we have some other needs? These IPCs are very Intel specific and for these the adsp_clock/power_gating_lock() interface is ok, but we need to put these into some Intel specific corner of SOF code as well.

Only for performance measurements. The motivation for me to propose such API was to give the ability to disable and enable this feature in the runtime from the SOF side without subsequent changes to the zephyr. But in the end it can be limit only to Kconfig.